.github/workflows/protobuf.yml

```diff
@@ -0,0 +1,83 @@
+name: protobuf
+# CI tests to run against the protobuf schema on change:
+
+on:
+  # run for any PRs raising changes to the protobuf files or setup
+  pull_request:
+    paths:
+      - 'cylc/flow/network/protobuf/**'
+
+jobs:
+  protobuf:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          # we need all of the commits on the PR branch in order to be able to add new ones
+          fetch-depth: 100
+
+      - name: Configure git
+        uses: cylc/release-actions/configure-git@v1
+
+      - name: Install Protobuf
+        uses: mamba-org/setup-micromamba@v1
+        with:
+          # install protobuf into a mamba env (note use shell = "bash -el {0}"
+          # to access this envionment)
+          environment-name: protobuf
+          create-args: protobuf
+          init-shell: bash
+
+      - name: Install bufbuild/buf
+        run: |
+          eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"  # activate homebrew
+
+          # NOTE: bufbuild does exist on conda-forge but hasn't been updated for a while
+          brew install bufbuild/buf/buf
+
+      - name: Lint
+        run: |
+          # lint .proto files
+          eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"  # activate homebrew
+          cd cylc/flow/network/protobuf
+          buf lint
```

Comment:
(Looks like `buf` can be installed via npm which might be easier to use in GH Actions, rather than activating homebrew every step)

---

<details><summary>Btw, first result when I googled "buf" 😆</summary>

<img width="289" alt="image" src="https://github.com/user-attachments/assets/6110d699-1894-4a9e-a5a1-8d57c62788d3">

</details>

Reply:
I wouldn't have used a system package manager if I could have avoided it (I don't like system package managers), but I can't remember.